Initial SST docs (#2478)
withastro/docs: src/pages/en/guides/deploy/sst.mdx

---
title: Deploy your Astro Site to AWS with SST
description: How to deploy your Astro site to AWS with SST
layout: ~/layouts/DeployGuideLayout.astro
i18nReady: true
---

You can deploy an Astro site using [SST](https://sst.dev), an open-source framework for deploying applications to AWS. This is the easiest way to get a fully serverless Astro deployment on AWS with SSR support.

> You can also use any additional SST constructs like Cron Jobs, Buckets, Queues, etc while maintaining typesafety.

Reviewer: We are using "type-safety" across the docs rather than typesafety or "type safety", I'd change this to the former for consistency.
```suggestion
You can also use any additional SST constructs like Cron Jobs, Buckets, Queues, etc while maintaining type-safety.
```

## Quickstart

1. Create an astro project

> 1. Run `npx create-sst@rc`

Author: Yep we'll come back to update this once we're released

1. It should detect that you are using Astro and ask you to confirm.
1. Once you're ready for deployment you can run `npx sst deploy --stage=production`

> If you wish to use any [additional SST constructs](https://docs.sst.dev/) they can be added to `sst.config.ts`. Then take a look at [Resource Binding](https://docs.sst.dev/resource-binding) to understand how to use them in your Astro code.

Reviewer 2: `````suggestion
To use any [additional SST constructs](https://docs.sst.dev/), add them to `sst.config.ts`. 

```js
// config example here might be nice?
```

Consult the [SST docs on Resource Binding](https://docs.sst.dev/resource-binding) to understand how to use them in your Astro code.
`````

What would you think about adding an example of what the configuration file should look like with contructs added?

> Or if you prefer a walkthrough you can watch [this video](https://www.youtube.com/watch?v=AFP3ZHxO7Gg).

Reviewer 2: ```suggestion
You can also watch [a video walkthrough of this process](https://www.youtube.com/watch?v=AFP3ZHxO7Gg) that will guide you through the steps.
```
We like to try to make link text as descriptive as possible!

> If you have any questions [ask in the SST discord](https://discord.gg/sst).

Reviewer 2: ```suggestion
If you have any questions, you can [ask in the SST discord](https://discord.gg/sst).
```

layout: ~/layouts/DeployGuideLayout.astro
i18nReady: true
---

You can deploy an Astro site using [SST](https://sst.dev), an open-source framework for deploying fully serverless applications to AWS with SSG and SSR support.

You can also use any additional SST constructs like Cron Jobs, Buckets, Queues, etc while maintaining type-safety.

## Quickstart

1. Create an astro project
1. Run `npx create-sst@rc`
1. It should detect that you are using Astro and ask you to confirm.
1. Once you're ready for deployment you can run `npx sst deploy --stage=production`

To use any [additional SST constructs](https://docs.sst.dev/), add them to `sst.config.ts`. 

```ts {2}
app.stack(function Site(ctx) {
  const bucket = new Bucket(ctx.stack, "public");
  const site = new AstroSite(ctx.stack, "site", {
    bind: [bucket],
  });

  ctx.stack.addOutputs({
    url: site.url,
  });
});
```

Consult the [SST docs on Resource Binding](https://docs.sst.dev/resource-binding) to understand how to use them in your Astro code.

You can also watch [a video walkthrough of this process](https://www.youtube.com/watch?v=AFP3ZHxO7Gg) that will guide you through the steps.

> If you have any questions, you can [ask in the SST discord](https://discord.gg/sst).

Reviewer 2: ```suggestion

You can also watch [a video walkthrough of this process](https://www.youtube.com/watch?v=AFP3ZHxO7Gg) that will guide you through the steps.

### SST constructs
To use any [additional SST constructs](https://docs.sst.dev/), add them to `sst.config.ts`. 

```ts {2}
app.stack(function Site(ctx) {
  const bucket = new Bucket(ctx.stack, "public");
  const site = new AstroSite(ctx.stack, "site", {
    bind: [bucket],
  });

  ctx.stack.addOutputs({
    url: site.url,
  });
});
```

Consult the [SST docs on Resource Binding](https://docs.sst.dev/resource-binding) to understand how to use them in your Astro code.

If you have any questions, you can [ask in the SST Discord](https://discord.gg/sst).
```
Ah! Had a last minute change of heart checking the preview! 

- capitalize Discord
- Walkthrough after the steps that it actually walks through?
- A header to separate SST contructs visually

Any concerns here?

---
title: Deploy your Astro Site to AWS with SST
description: How to deploy your Astro site to AWS with SST
layout: ~/layouts/DeployGuideLayout.astro
i18nReady: true
---

You can deploy an Astro site using [SST](https://sst.dev), an open-source framework for deploying fully serverless applications to AWS with SSG and SSR support.

You can also use any additional SST constructs like Cron Jobs, Buckets, Queues, etc while maintaining type-safety.

## Quickstart

1. Create an astro project
1. Run `npx create-sst@rc`
1. It should detect that you are using Astro and ask you to confirm.
1. Once you're ready for deployment you can run `npx sst deploy --stage=production`

To use any [additional SST constructs](https://docs.sst.dev/), add them to `sst.config.ts`. 

```ts {2} {4}
app.stack(function Site(ctx) {
  const bucket = new Bucket(ctx.stack, "public");
  const site = new AstroSite(ctx.stack, "site", {
    bind: [bucket],
  });

  ctx.stack.addOutputs({
    url: site.url,
  });
});
```

> And then access them in your astro code
> 
> ```ts
> import { Bucket } from "sst/node/bucket"
> console.log(Bucket.public.bucketName)
> ```

Reviewer 2: ```````suggestion
And then access them in your astro code

```astro
---
import { Bucket } from "sst/node/bucket"
console.log(Bucket.public.bucketName)
---
```
``````
If this is a `.astro` file, you can use `astro` as the code language. Slight preference for throwing the code fences in there, too. But, otherwise, approved!

---
title: Deploy your Astro Site to AWS with SST
description: How to deploy your Astro site to AWS with SST
layout: ~/layouts/DeployGuideLayout.astro
i18nReady: true
---

You can deploy an Astro site using [SST](https://sst.dev), an open-source framework for deploying fully serverless applications to AWS with SSG and SSR support.

You can also use any additional SST constructs like Cron Jobs, Buckets, Queues, etc while maintaining type-safety.

## Quickstart

1. Create an astro project
1. Run `npx create-sst@rc`
1. It should detect that you are using Astro and ask you to confirm.
1. Once you're ready for deployment you can run `npx sst deploy --stage=production`

You can also watch [a video walkthrough of this process](https://www.youtube.com/watch?v=AFP3ZHxO7Gg) that will guide you through the steps.

### SST constructs
To use any [additional SST constructs](https://docs.sst.dev/), add them to `sst.config.ts`. 

> ```ts {2} {4}

Reviewer 2: ```suggestion
```ts {2} {4} title="sst.config.ts"
```

app.stack(function Site(ctx) {
  const bucket = new Bucket(ctx.stack, "public");
  const site = new AstroSite(ctx.stack, "site", {
    bind: [bucket],
  });

  ctx.stack.addOutputs({
    url: site.url,
  });
});
```

> And then access them in your astro code

Reviewer 2: ```suggestion
And then access them in your `.astro` file.
```
